providers/azure/virtual_network.go

```diff
@@ -32,6 +32,13 @@ func (g VirtualNetworkGenerator) createResources(ctx context.Context, iterator n
 	var resources []terraformutils.Resource
 	for iterator.NotDone() {
 		virtualNetwork := iterator.Value()
+		tferName := terraformutils.TfSanitize(*virtualNetwork.Name)
+		for _, resource := range resources {
+			if tferName == resource.ResourceName {
+				*virtualNetwork.Name = *virtualNetwork.Name + "_" + GenerateRandomString(6)
```

Comment:
You can use virtual network ID instead random string.
Random string return different results in each iteration. Some people use terraformer for detect diff/changes so random string it's not good for them.

Reply:
@sergeylanzman - I have made the changes. Please approve it.